### DISCUSSION TITLE
Ideas of enhancements for persist function

### DISCUSSION THREAD

**Discussion by AkariiinMKII** [ScoopInstaller/Scoop#5169] [Ideas]
Here's the situation I met when using `persist` function in manifest. 
There is an app, whose config file needs to be persisted but:
- The config file won't be generated before launching.
- The app is not compatible with empty config file (may cause app broken).

Additionally, all `persist` values without existing items found are treated as folder name in current version. 
Of course we can use custom scripts to solve this, however, it will be nice if we are able to declare properties like `persist_as_file` and `force_persist` in `persist` function, and it is supposed to work when installing, uninstalling and updating:

![installation](https://user-images.githubusercontent.com/6019344/193328930-bff13b37-a8d3-4c69-9142-6adf9dede728.jpg)

![uninstallation](https://user-images.githubusercontent.com/6019344/193328949-e03e607d-5281-4876-808f-7d4a517ba4df.jpg)

Furtherly, some apps may break file links that `persist` cannot work with them, could we add a `backup` field in manifest with functions to backup and restore items instead of persisting, which is also implemented by custom scripts.

**Comment by niheaven** [comment]
See #3248 for persistence rework, but that’s outdated and needs some updating. Does that logic fit your need?

**Reply by AkariiinMKII** [reply]
Yes, it perfectly meets my demands if add a new field `force`.

**Reply by niheaven** [reply]
Okay, then I'll carefully cherrypick the PR's commit and update it, please wait for it.
